virttest/qemu_vm.py

```diff
@@ -4209,17 +4224,10 @@ def get_block_old(self, blocks_info, p_dict={}):
                     return block.split(":")[0]
         else:
             for block in blocks_info:
-                match = True
                 for key, value in six.iteritems(p_dict):
-                    if isinstance(value, bool):
-                        check_str = "u'%s': %s" % (key, value)
-                    else:
-                        check_str = "u'%s': u'%s'" % (key, value)
-                    if check_str not in str(block):
-                        match = False
-                        break
-                if match:
-                    return block['device']
+                    for (k, v) in traverse_nested_dict(block):
+                        if (k == key) and (value == v or value in v):
```

Comment:
Should not have `value in v` because it may get the wrong item.
For example, the vm has the following two block devices, and you want to find device id by the filename `image1`,
```
block100: filename="image100"
block1: filename="image1"
```
finally, it would return `block100` because its filename does contain the pattern `image1`.